azure/compute/vm.go

```diff
@@ -0,0 +1,153 @@
+		Length:  pulumi.Int(20),
+		Special: pulumi.Bool(true),
+	})
+	if err != nil {
+		return nil, nil, nil, pulumi.StringOutput{}, err
+	}
+
+	windowsOsProfile := compute.OSProfileArgs{
+		ComputerName:  pulumi.String(name),
+		AdminUsername: pulumi.String(adminUsername),
+		AdminPassword: windowsAdminPassword.Result,
+		CustomData:    userData,
+	}
+
+	if firstLogonCommand != nil {
+		windowsOsProfile.WindowsConfiguration = compute.WindowsConfigurationArgs{
+			AdditionalUnattendContent: compute.AdditionalUnattendContentArray{
+				compute.AdditionalUnattendContentArgs{
+					ComponentName: compute.ComponentNames_Microsoft_Windows_Shell_Setup,
+					PassName:      compute.PassNamesOobeSystem,
+					SettingName:   compute.SettingNamesFirstLogonCommands,
+					Content:       firstLogonCommand,
+				},
+			},
+		}
+	}
+
+	vm, publicIP, nw, err := newVMInstance(e, name, imageUrn, instanceType, windowsOsProfile)
+	if err != nil {
+		return nil, nil, nil, pulumi.StringOutput{}, err
+	}
+
+	return vm, publicIP, nw, windowsAdminPassword.Result, nil
+}
+
+func newVMInstance(e azure.Environment, name, imageUrn, instanceType string, osProfile compute.OSProfilePtrInput) (*compute.VirtualMachine, *network.PublicIPAddress, *network.NetworkInterface, error) {
+	vmImageRef, err := parseImageReferenceURN(imageUrn)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	publicIP, err := network.NewPublicIPAddress(e.Ctx, e.Namer.ResourceName(name), &network.PublicIPAddressArgs{
+		PublicIpAddressName:      e.Namer.DisplayName(pulumi.String(name)),
+		ResourceGroupName:        pulumi.String(e.DefaultResourceGroup()),
+		PublicIPAllocationMethod: pulumi.String(network.IPAllocationMethodStatic),
+		Tags:                     e.ResourcesTags(),
+	})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	nwInt, err := network.NewNetworkInterface(e.Ctx, e.Namer.ResourceName(name), &network.NetworkInterfaceArgs{
+		NetworkInterfaceName: e.Namer.DisplayName(pulumi.String(name)),
+		ResourceGroupName:    pulumi.String(e.DefaultResourceGroup()),
+		IpConfigurations: network.NetworkInterfaceIPConfigurationArray{
+			network.NetworkInterfaceIPConfigurationArgs{
+				Name: e.Namer.DisplayName(pulumi.String(name)),
+				Subnet: network.SubnetTypeArgs{
+					Id: pulumi.String(e.DefaultSubnet()),
+				},
+				PrivateIPAllocationMethod: pulumi.String(network.IPAllocationMethodDynamic),
+				PublicIPAddress: network.PublicIPAddressTypeArgs{
+					Id: publicIP.ID(),
+				},
+			},
+		},
+		Tags: e.ResourcesTags(),
+	})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	vm, err := compute.NewVirtualMachine(e.Ctx, e.Namer.ResourceName(name), &compute.VirtualMachineArgs{
+		ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+		VmName:            e.Namer.DisplayName(pulumi.String(name)),
+		HardwareProfile: compute.HardwareProfileArgs{
+			VmSize: pulumi.StringPtr(instanceType),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
If `instanceType` is used to set the `VmSize`, should it be called `instanceSize` instead ?

Reply:
So Azure calls it `VmSize` but it really is the `instanceType`, it's really the equivalent of the AWS `instanceType` and not just the # of CPUs.